hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -106,10 +106,8 @@ int64_t MSKTOP(int64_t value) {
 vixl::MemOperand M(Vptr p) {
   assertx(p.base.isValid());
   if (p.index.isValid()) {
-    assertx(p.disp == 0);
```

Comment:
I'll revisit the change I was probably too quick to remove it.  I think your lowering question might be related to this pull request.
https://github.com/facebook/hhvm/pull/7058